genisys/make_password.py

```diff
@@ -15,6 +15,11 @@ def make_password(self, raw_password, filename):
         secret_password = pbkdf2_sha256.hash(raw_password)
         self.config["users"]["root-login"] = secret_password
 
+        # store updated configuration back into config file
+        parser = YAMLParser
+        config = parser.as_dict(filename)
```

Comment:
Take another look at the implementation of YAMLParser. As you have it here, `parser` becomes a reference to the class `YAMLParser` rather than an instance of the class. You need to do `YAMLParser(filename)` to call the YAMLParser's constructor, then you can do `parser.as_dict()` since the `self` parameter is implied for an instance method.